src/main/java/io/nats/client/impl/ByteArrayBuilder.java

```diff
@@ -0,0 +1,120 @@
+// Copyright 2015-2020 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.client.impl;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+
+public class ByteArrayBuilder {
```

Comment:
not sure. do we specifically want to hide it? It could be a useful utility for a dev

Reply:
Either way...

Reply:
I think we can just get rid of this class, it seems to be unnecessary and inefficient. If you do the string->bytebuffer conversions prior to serialization(maybe in something like the `NatsMessage` `Builder`) you then don't have to resize the serialized buffers at all as computing the buffer sizes is straightforward for bytebuffers unlike strings, but sizing the buffers properly is somewhat function specific so this might not be able to be fully generalized.